### ISSUE TITLE
MacOs

### ISSUE THREAD

**Issue by YuryKonvpalto** [msqr1/Vosklet#13]
Hi msqr1.
I have tested simple app with mic on Android phone - it loads well and works.
On the other hand, I tried to run app on iphone (tried both Safari and Chrome) and it seems something is blocking the model to be downloaded to iphone or couldnt be started. Its stucks on (model = await module.createModel(......)
So the same code works fine with android (Galaxy phone with chrome).

Seems like iphone's private file system works somehow different, or maybe closed at all for downloads,

Do you have ny ideas gow to circumvent it?

**Comment by msqr1** [comment]
I don't think the OS matters, what browser are you using?

**Comment by msqr1** [comment]
What version of Chrome/Safari are you using?

**Comment by YuryKonvpalto** [comment]
Chrome - 128.0
Safari - 17.6.1

**Comment by msqr1** [comment]
Hmm I can't reproduce it, is there an error in the console?

**Comment by msqr1** [comment]
I'm using createWritable to write in OPFS, having like 75% browser support (bad, idk why I did this). 
I am working to use SyncAccessHandle in a worker instead that can do the same thing, but with 92% browser support

**Comment by YuryKonvpalto** [comment]
Yes, maybe that was the reason it didnt work. Anyway - a greate work, Msqr1, I like your variant of Vosk very much, thank you for sharing your creation!

**Comment by msqr1** [comment]
I used SyncAccessHandle in a separate worker, which works but will create an extra worker for OPFS. I tried my best to integrate with the WASM workers from Emscripten, but it's literally impossible. 

Could you verify that this is fixed in the latest commit before I tag a new version? @YuryKonvpalto

**Comment by msqr1** [comment]
I don't have any Apple Device :(

**Comment by YuryKonvpalto** [comment]
Hi Msqr1! Sorry for a delay with replay, - have had some busy days.
Very intersning to check if it will do, - I'll try new version tonight and will ypu give a feedback on results.

**Comment by YuryKonvpalto** [comment]
I have checked it on iphone SE2020 and iphone 14 - even your exaple with mic doesnt work. I think its related to the Safari and iphone itself, - how they are managing browser filesystem. I have a chrome on iphone, but it fails either (iphone puts some restrictions on external browsers too).
In iphone you even cant see consol.log or put messages from console to textarea of html. Only if you have a mac (which I dont) you can see console logs.

I can tell you also, that I've tried vosk-browser - it works on iphone, because probably it loads models to cache.

**Comment by msqr1** [comment]
Vosk browser uses IndexedDB to store models, which has better support than OPFS, but slower read/write.

Could you tell me how you test HTML on an iPhone? How do you make Safari render a local HTML file?

**Comment by YuryKonvpalto** [comment]
![изображение](https://github.com/user-attachments/assets/ef16d5f9-4b29-484e-a3fe-f071db2fb60b)

This is how it lokks like on the desktop computer (Windows)

**Comment by YuryKonvpalto** [comment]
There are a inscriptoion in the textfield - "model is loaded". It appears as soon as model is loaded to the comp.

On iphone there is an inscription "model tries to load itself" - it appears before await module.createModel("./....) - Look the code below:

    module = await loadVosklet()

    htmlTextArea.innerHTML = 'model tries to load itself';

    let modelRu = await module.createModel("./vosk-model-small-ru-0.4.tar.gz", "model", "ru")
    
    recognizer = await module.createRecognizer(modelRu, 16000)

    if (modelRu) {
        htmlTextArea.innerHTML = 'model is loaded';
    }


So, if modelRu succesfully has been loaded, it appears "model is loaded" inthe textfield. And its really happens on the desktop.
On iphone it stucks on "model tries to load itself".

For testing I use ngrok.
The ngrok tunnel works well on desktop - I checked it.
Dont know if it works on android though.

**Comment by msqr1** [comment]
Could you check once more? I switched to use Cache API

**Comment by msqr1** [comment]
Can you confirm please 🙏? I'm waiting on your feedback to release a new version! @YuryKonvpalto

**Comment by YuryKonvpalto** [comment]
Hi msqr1,

it works very well on Desktop (end even maybe a bit faster, atleast thats like it seems).
I have tested it on iphone, it stucks here:

it prints - "test" - so before loading a model it looks fine,
but never prints "model tries to load itself"

Look at code below.
I'll check it more closer tonight (do some test), but Apple made it very hard to debug code without desktop-Mac.


     htmlTextArea.innerHTML = 'test';
    module = await loadVosklet()
   htmlTextArea.innerHTML = 'model tries to load itself';

**Comment by msqr1** [comment]
I ported kaldi with openBLAS (original approach), which increased recognition performance by around 20% since openBLAS is faster than the  reference one.

Thank you for staying with me after all of this. 

There is no way the Cache API doesn't work on iOS. 😭

I tested, and it worked on a Mac and Android, a Windows, on Firefox, Edge, and Chrome and additionally on Safari for the Mac. Literally everything except iOS
